nixos/modules/services/networking/wpa_supplicant.nix

```diff
@@ -68,14 +74,27 @@ let
       wantedBy = [ "multi-user.target" ];
       stopIfChanged = false;
 
-      path = [ package ];
+      path = [ package pkgs.gawk ];
+      serviceConfig.RuntimeDirectory = "wpa_supplicant";
+      serviceConfig.EnvironmentFile = mkIf (cfg.environmentFile != null)
+        (builtins.toPath cfg.environmentFile);
```

Comment:
`toPath` is [deprecated](https://nixos.org/manual/nix/stable/#builtin-toPath)

Reply:
Actually, why are you using `toPath` at all?

Reply:
Good question: I think I mean to write `toString`, in case `environmentFile` is a path, to avoid pulling it into the Nix store.